File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/1299. Replace Elements with Greatest Element on Right Side/C++ solution/1299. Replace Elements with Greatest Element on Right Side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n = arr.size();
        int maxval = arr[n -1];
        vector<int> new_arr(n);
        new_arr[n-1] = -1;

        for (int i = n - 2; i >= 0; i--)
        {
            maxval = max(maxval, arr[i + 1]);
            new_arr[i] = maxval;
        }

        return new_arr;
    }
};

File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/1299. Replace Elements with Greatest Element on Right Side/Python solution/1299. Replace Elements with Greatest Element on Right Side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        new_arr = [0] * len(arr)
        maxval = new_arr[-1]
        new_arr[-1] = -1

        for i in range(len(arr) - 2, -1, -1):
            print(i)
            maxval = max(maxval, arr[i + 1])
            new_arr[i] = maxval
        
        return new_arr

